### DISCUSSION TITLE
Strongswan Performance - Time taken by charon daemon to establish more than 4k tunnels too high

### DISCUSSION THREAD

**Discussion by deeptishm** [strongswan/strongswan#1469] [Q&A]
Following are our observations related to Charon performance on 256 core HP AMD Server with 128 GB RAM with 100G NIC:

- 512 Tunnels - ~2mins
- 1024 Tunnels - ~3mins
- 2048 Tunnels - 4-5mins
- 4096 Tunnels - ~12-13mins
- 8192 Tunnels - 20-24 mins
- 16K tunnels - ~1 hr
- 32k Tunnels - Not working.

Can you please throw some light on why the Charon is taking so much time for tunnel establishment with 0 traffic load on such a high performing server?

**Comment by tobiasbrunner** [comment]
That really depends on a lot of things, e.g. the connection config (authentication method, DH exchange) or tuning the daemon itself (number of threads, [hash table](https://docs.strongswan.org/docs/5.9/config/lookupTuning.html), [job priority](https://docs.strongswan.org/docs/5.9/config/jobPriority.html) etc.).
